rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,295 @@
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+### Trip dropped
+This removes a trip (either added or scheduled), indicating that it will not be running.
+
+Event type: `com.mbta.ctd.glides.trip-dropped.v1`
+Fields in the event:
+- `actor` (Actor): the inspector adding the trip
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+- `reason` (string): a text description of why the trip is being dropped
+### Trip restored
+This restores a trip (either added or scheduled) that was previously dropped.
+
+Event type: `com.mbta.ctd.glides.trip-restored.v1`
+Fields in the event:
+- `actor` (Actor): the inspector adding the trip
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+### Trip updated
+This indicates that a trip was updated in some fashion:  consist, operators, departure time. Fields which are not present are not considered to be updated.
+
+Event type: `com.mbta.ctd.glides.trip-updated.v1`
+Fields in the event:
+- `actor` (Actor): the inspector adding the trip
+- `tripKey` (Trip Key): which trip is being updated
+- `departureTime` (HH:MM:SS time relative to the service day, optional): if present, the new time that the train is expected to depart
+- `consist` (array of Optional Car, optional): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist. The front car should be listed first.
+- `operators` (array of Optional Operator, optional): the list of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+## Long-term storage / querying
+For future use, events will be archived to S3, using Kinesis Firehose. They can either be sent to LAMP's folder, or a new folder.
+
+# Reference-level explanation
+## Effects on RTR
+These inputs should be similar to what RTR is already receiving from OCS.
+- Trip added: `TSCH NEW`
+- Trip dropped: `TSCH DEL` with delete status 1
+- Trip restored: `TSCH DEL` with delete status 0
+- Trip updated: `TSCH CON` (set consist), `TSCH OFF` (change departure time)
+
+## Examples
+### Train departing 15 minutes later than scheduled
+- Inspector Alice (badge number: 123) is working at Mattapan
+- The 1:30am trip is being held 15 minutes to wait for a final Red Line train
+```json
+{
+  "specversion": "1.0",
+  "type": "com.mbta.ctd.glides.trip-updated.v1",
+  "source": "glides.mbta.com",
+  "id": "19fdb184-7dd6-4664-8472-04bd6177ec44",
+  "time": "2023-01-23T01:25:00-05:00",
+  "data": {
+    "actor": {
+      "username": "ainspector",
+      "badgeNumber": "123"
+    },
+    "tripKey": {
+      "serviceDate": "2023-01-22",
+      "scheduledTime": "25:30:00"
```

Comment:
Is this enough to identify a trip? Wouldn't the line, direction, and departure station also be needed, or some other unique ID?

Reply:
Good point; it should include the origin station as well.